Assets/Scripts/Layers/LayerTypes/LayerData.cs

```diff
@@ -205,5 +205,22 @@ public void RemoveProperty(LayerPropertyData propertyData)
             layerProperties.Remove(propertyData);
             PropertyRemoved.Invoke(propertyData);
         }
+
+        /// <summary>
+        /// Recursively collect all assets from each of the property data elements for loading and saving
+        /// purposes. 
+        /// </summary>
+        /// <returns>A list of assets on disk</returns>
+        public IEnumerable<Asset> GetAssets()
+        {
+            var assetsOfCurrentLayer = layerProperties
+                .OfType<ILayerPropertyDataWithAssets>()
+                .SelectMany(p => p.GetAssets());
+
+            var assetsOfAllChildLayers = children
+                .SelectMany(l => l.GetAssets());
+
+            return assetsOfAllChildLayers.Concat(assetsOfCurrentLayer);
```

Comment:
Dit geeft niet recursive alle assets, alleen die van jezelf en je directe children

Reply:
En omdat de directe kinderen van hun directe kinderen dat ook weer doen krijg je recursiviteit :)

Reply:
Ah, mijn fout ik dacht dat je op de children ILayerPropertyDataWithAssets.GetAssets() callde en niet LayerData.GetAssets().

Overigens is het misschien beter om deze hele functie in RootLayer te zetten, of is er een scenario waar je de assets van 1 specifieke layer branch zou willen opvragen?

Reply:
Door het in LayerData te zetten krijg ik recursiviteit, en de winst dat code van 1 branch wat kan opvragen. Als ik het in RootLayer zet dan moet ik de het deep bij elkaar sprokkene allemaal in RootLayer doen en ik zag geen meerwaarde daarin; sterker nog: alleen maar een complexere functie :)